platforms/core-runtime/service-provider/src/main/java/org/gradle/internal/service/ServiceProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.service;
+
+public interface ServiceProvider {
```

Comment:
💅 `ServiceRegistrationProvider` (since these are passed to `ServiceRegistration#add` and provide not the services themselves but the reflection based registration of services).

Reply:
Alright, makes sense. Let's go with `ServiceRegistrationProvider` then.